### DISCUSSION TITLE
Question on VRAM usage (differences in AMD and NVIDIA)

### DISCUSSION THREAD

**Discussion by lbeltrame** [vladmandic/sdnext#2614] [Q&A] [answered]
I'm currently comparing SD local generations on an AMD RX7800XT (16G VRAM) w/ROCm and one of the offerings from Paperspace, a Quadro RTX A4000 (also 16G VRAM). What I don't seem to understand is that there is a massive difference in VRAM usage using the same models and LoRAs. 

<strike>The former, with a slightly less upscale value, starting from the same resolution, uses approximately 6-8G VRAM. The latter gets very close to the limit (15.8 out of 16). The AMD card uses sub-quadratic attention as optimization (and I had to use VAE with fp32 as there are some overflows otherwise), while the NVIDIA card uses xformers + flash attention (with SDP I would get over 16G used, but I admit I use quite large generations)

Checkpoints and LoRAs used on both GPUs are identical, and often identical prompts. I don't mind if this is actually expected, but it was striking. Where should I look to find more clues?

Both cards use pytorch 2.1 with the respective backends (ROCm 5.6 and CUDA 12.1); the AMD installation uses Python 3.11, NVIDIA Python 3.10. Save for the pytorch and related dependencies, the Python packages installed are identical..</strike>

No need for that, see below.

**Comment by vladmandic** [comment]
app settings, actual numbers, image params?

**Reply by lbeltrame** [reply, marked as answer]
PEBKAC, sorry for the noise. I reviewed the logs now and it looks the lines were from an earlier experiment on AMD with a much lower resolution. Therefore, from an accurate review, memory usage looks very similar (~100Mb difference).
